draft-gpew-priv-ppm.md

```diff
@@ -1136,28 +1151,26 @@ struct {
 The sequence of Transition messages corresponds to the ReportShare sequence of
 the AggregateInitReq. The order of these sequences MUST be the same (i.e., the
 nonce of the first Transition MUST be the same as first ReportShare and so on).
-`tag` is an HMAC-SHA256 tag over the serialized message, excluding the `tag`
-field itself, computed using the `agg_auth_key` shared by the aggregators.
-The helper's response to the leader is an HTTP 200 OK whose body is the
-AggregateResp.
+The tag `tag` field is an HMAC-SHA256 tag over the serialized message, excluding
+the `tag` field itself, computed using the `agg_auth_key` shared by the
+aggregators. The helper's response to the leader is an HTTP 200 OK whose body is
+the AggregateResp.
 
-After verifying the authentication tag, the leader handles the AggregateResp as
-follows. It first checks that the sequence of Transition messages corresponds to
-the ReportShare sequence of the AggregateInitReq. If the messages appear out of
-order, any are missing, or a nonce is not recognized, then the leader MUST abort
-and alert its peer with error "XXX".
+The leader handles the AggregateResp as follows. It first verifies the tag. If
+this check fails, it MUST abort with error "invalidHmac".
+
+Next, it first checks that the sequence of Transition messages corresponds to
+the ReportShare sequence of the AggregateInitReq. If any message appears out of
+order or is missing, or a transition has an unrecognized nonce, then the leader
```

Comment:
I believe that "or is missing" conflicts with section 4.4.4.1, assuming a multi-round VDAF. That section says that the leader will filter out reports that correspond to a state of FAILED, in which case the next AggregateContinueReq would have fewer transitions/nonces/reports than the AggregateInitReq and AggregateResp that came before it.

Reply:
Hmm, maybe we just need to sharpen this a little. This paragraph pertains to the leader processing `AggregateResp`. The helper is not allowed to change the sequence of nonces relative to the previous `AggregateReq`. In particular, if a nonce is missing (relative to the previous request), then we abort.